_posts/2022-10-28-multiple-data-source.md

```diff
@@ -0,0 +1,97 @@
+* **Usability**:
+  * As someone who is responsible for managing employee access to sensitive data, no longer need to manage credentials across several clusters and configure, install, provision, admin multiple OpenSearch Dashboards.
+  * As someone who builds reports across multiple teams, no longer need to navigate across multiple OpenSearch Dashboards applications in order to see all data needed, or manually build reports combining the data from different sources.
+* **Flexibility**: Customer could add and connect to any compatible data sources, and remove them anytime.
+
+###What can you do with multi data source experimental launch?
+
+The first phase of multiple data source will provide OpenSearch Dashboards ability to query data from multiple OpenSearch endpoints with compatible versions, with this feature you can:
+
+* Add/remove the data sources which is compatible with Elasticsearch DSL, e.g. AWS OpenSearch Domains, AWS OpenSearch Serverless collections, and customer self-managed OpenSearch clusters
+* Create visualizations that renders data from different index patterns to compare time-series data in the same dashboard
+
+![Use case by persona](/assets/media/blog-images/2022-10-28-multiple-data-source/usecase-by-persona.png){: .img-fluid}
+
+
+###How does multi data source feature work?
+
+
+**High-level architecture diagram**
+
+![Legacy architecture](/assets/media/blog-images/2022-10-28-multiple-data-source/legacy.png){: .img-fluid}
+![Evolution architecture](/assets/media/blog-images/2022-10-28-multiple-data-source/evolution.png){: .img-fluid}
+
+Multi data source feature introduces a new saved object data-source to describe data source connection, including endpoint, authentication information, etc. Users can dynamically add data source in OpenSearch Dashboards through UI or API.
+
+When using this feature, index pattern which are associated with data source will have reference field filled with:
+
+![IndexPattern with dataSource](/assets/media/blog-images/2022-10-28-multiple-data-source/indexpattern-with-ds.png){: .img-fluid}
+
+Current supported data source saved object attributes (pseudo code):
+
+```
+DataSourceAttributes extends SavedObjectAttributes {
+    title,
+    description,
+    endpoint,
+    auth: {
+      type,
+      credentials,
+    };
+}
+```
+
+In supporting other OpenSearch Dashboards’ components to interact with user data source cluster, this feature forked and repurposed existing `opensearch_service` to allow making a data source connection. When OpenSearch Dashboards needs to access data sources on behalf of the user, it will fetch the data source info from its metadata storage by data source id, then send the request to the corresponding data source endpoint.
```

Comment:
Suggested rewrite:

To enable other Dashboards components to interact with a user data source cluster, this feature forked and repurposed the existing `opensearch_service` to create a data source connection. When Dashboards needs to access data sources, it will fetch the data source information from its metadata storage by data source id and then send the request to the corresponding data source endpoint.